pkg/router/f5/f5.go

```diff
@@ -525,11 +556,13 @@ func (f5 *f5LTM) ensureVserverHasPolicy(vserverName, policyName string) error {
 	glog.V(4).Infof("Checking whether vserver %s has policy %s...",
 		vserverName, policyName)
 
+	vserverUriPath := f5.iControlVserverUriPath(vserverName)
+
 	// We could use fmt.Sprintf("https://%s/mgmt/tm/ltm/virtual/%s/policies/%s",
 	// f5.host, vserverName, policyName) here, except that F5 iControl REST
 	// returns a 200 even if the policy does not exist.
 	vserverPoliciesUrl := fmt.Sprintf("https://%s/mgmt/tm/ltm/virtual/%s/policies",
-		f5.host, vserverName)
+		f5.host, vserverUriPath)
```

Comment:
It seems odd and slightly convoluted that we call `iControlVserverUriPath`—which gets the partition from `f5` and uses `strings.Split`, `path.Join`, and `escapeiControlUriPath` to compose and escape the resource path component—and then we call `fmt.Sprintf`—where we tell it the hostname and resource path component to interpolate.  Perhaps it would be clearer to provide `prependWithPartition` and `prependWithTopLevelPartition` methods and let callers explicitly prepend and then encode as separate steps. Might need to think about it.